### PR TITLE
Update MRI to 2.3.7

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: lib/truffle/stringio.rb [hunk 1]
```diff
@@ -567,7 +567,10 @@ def sysread(length=nil, buffer='')
   end
 
   alias_method :readpartial, :sysread
-  alias_method :read_nonblock, :sysread
+  
+  def read_nonblock(length, buffer='', **options)
```
Comment (eregon) [inline lib/truffle/stringio.rb]: This could just be declared as a keyword argument like `, exception: true`.

File: lib/mri/forwardable.rb [hunk 2]
```diff
@@ -187,9 +184,6 @@ def def_instance_delegator(accessor, method, ali = method)
       def #{ali}(*args, &block)
         begin
           #{accessor}.__send__(:#{method}, *args, &block)
-        rescue ::Exception
-          $@.delete_if{|s| ::Forwardable::FILE_REGEXP =~ s} unless ::Forwardable::debug
-          ::Kernel::raise
```
Comment (eregon) [inline lib/mri/forwardable.rb]: What's problematic here?

Comment (chrisseaton) [reply lib/mri/forwardable.rb]: I don't think anything's problematic - what do you mean?

File: lib/truffle/stringio.rb (hunk 1 above)
Comment (chrisseaton) [reply lib/truffle/stringio.rb]: It seems to accept any arguments and ignore ones it doesn't know about - would that still work with explicit keywords?

```
irb(main):001:0> require 'stringio'
=> false
irb(main):002:0> StringIO.new('foo').read_nonblock(3, foo: 1)
=> "foo"
```

File: lib/mri/forwardable.rb (hunk 2 above)
Comment (eregon) [reply lib/mri/forwardable.rb]: I mean, why did you remove those lines? There were modified previously?

Comment (chrisseaton) [reply lib/mri/forwardable.rb]: Oh the equivalent rescues in the MRI implementation were deleted by them - the semantics changed.

File: lib/truffle/stringio.rb (hunk 1 above)
Comment (eregon) [reply lib/truffle/stringio.rb]: Interesting, it seems MRI indeed doesn't check it. So this is fine.
Indeed, adding the explicit keyword argument would prevent passing unknown kwargs.

File: src/main/c/openssl/ossl_ssl.c [hunk 3]
```diff
@@ -1441,10 +1441,9 @@ ossl_ssl_read_internal(int argc, VALUE *argv, VALUE self, int nonblock)
 	if (RSTRING_LEN(str) >= ilen)
 	    rb_str_modify(str);
 	else
-	    rb_str_modify_expand(str, ilen - RSTRING_LEN(str));
+	    rb_str_resize(str, ilen);
     }
     OBJ_TAINT(str);
-    rb_str_set_len(str, 0);
     if (ilen == 0)
```
Comment (chrisseaton) [inline src/main/c/openssl/ossl_ssl.c]: We need to comment this fix so I know to keep it when I import new OpenSSL changes for the next version. Why do we need this and MRI doesn't?

Thanks for the fix though!

File: test/truffleruby-tool/lib/activesupport_exclusions.yaml [hunk 4]
```diff
@@ -1,3 +1,6 @@
 ---
 TimeExtMarshalingTest:
 - test_marshalling_preserves_fractional_seconds
+FileStoreTest:
+- test_filename_max_size
```
Comment (nirvdrum) [inline test/truffleruby-tool/lib/activesupport_exclusions.yaml]: What changed in 2.3.7 to necessitate this?

Comment (chrisseaton) [reply test/truffleruby-tool/lib/activesupport_exclusions.yaml]: I think the behaviour of `tempfile` changed, part of the security mitigations.

**Pull request merged**